hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -859,8 +860,10 @@ private void validateSchema() throws HoodieUpsertException, HoodieInsertExceptio
       Schema writerSchema = HoodieAvroUtils.createHoodieWriteSchema(config.getSchema());
       Schema tableSchema = HoodieAvroUtils.createHoodieWriteSchema(existingTableSchema.get());
       AvroSchemaUtils.checkSchemaCompatible(tableSchema, writerSchema, shouldValidate, allowProjection, getDropPartitionColNames());
+    } catch (SchemaCompatibilityException e) {
+      throw e;
     } catch (Exception e) {
-      throw new HoodieException("Failed to read schema/check compatibility for base path " + metaClient.getBasePath(), e);
+      throw new SchemaCompatibilityException("Failed to read schema/check compatibility for base path " + metaClient.getBasePath(), e);
```

Comment:
It looks like all such exception changes only make the exception thrown more specific, i.e., subclass of previous thrown exception (`SchemaCompatibilityException` class is a child of `HoodieException` class).  Make sure that's case, and if not, we need to add docs to explicitly explain the outliers, since users catching the previous exception type may miss the new one.